### PR TITLE
Update scripts for Flink ML 2.3.0 release

### PR DESCRIPTION
**Pull request by jiangxin369**
## What is the purpose of the change

Fix some bugs when preparing the release of 2.3.0.

## Brief change log

  - Replace the version with perl to avoid different behaviors of sed among different OS
  - Deploy artifacts of each Flink version

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: tools/releasing/deploy_staging_jars.sh [hunk 1]
```diff
@@ -41,7 +41,11 @@ fi
 
 cd ${PROJECT_ROOT}
 
-echo "Deploying to repository.apache.org"
-${MVN} clean deploy -Papache-release -DskipTests -DretryFailedDeploymentCount=10 $CUSTOM_OPTIONS
+FLINK_VERSIONS=("1.15" "1.16" "1.17")
```
Comment (lindong28) [inline tools/releasing/deploy_staging_jars.sh:44]: We probably don't want to modify this script every time we prepare a new Flink ML release.

Would it be simpler to use environment variables to pass versions and document the usages of this environment variables in the release wiki?

**Review by lindong28 — COMMENTED** [review]
Thanks for the PR. Left one comment below.

File: tools/releasing/deploy_staging_jars.sh [hunk 2]
```diff
@@ -22,6 +22,14 @@
 ##
 MVN=${MVN:-mvn}
 CUSTOM_OPTIONS=${CUSTOM_OPTIONS:-}
+SUPPORTED_FLINK_VERSIONS=${SUPPORTED_FLINK_VERSIONS:-}
+
+# The variable SUPPORTED_FLINK_VERSIONS must be set, which contains all supported flink
+# versions with comma separated, e.g. "1.15,1.16.1.17".
+if [ "${SUPPORTED_FLINK_VERSIONS}" = "" ]; then
```
Comment (jiangxin369) [inline tools/releasing/deploy_staging_jars.sh:29]: Here we depend on a variable `SUPPORTED_FLINK_VERSIONS`. @lindong28 Could you help update the release doc[1] to set this variable before running the script?

[1] https://cwiki.apache.org/confluence/display/FLINK/Creating+a+Flink+ML+Release#CreatingaFlinkMLRelease-Createareleasebranch

Comment (lindong28) [reply tools/releasing/deploy_staging_jars.sh:29]: Sure. I will do it.

**Comment by lindong28** [comment]
Thanks for the PR. LGTM.

**Pull request closed without merging**